x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -275,8 +323,10 @@ export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitializat
   router.delete(
     {
       path: '/api/ml/data_frame/analytics/{analyticsId}',
+      // validate: false,
       validate: {
         params: analyticsIdSchema,
+        query: analyticsIdIndexSchema,
```

Comment:
I wonder if rather than passing the name of the index in the query, it should instead pass a boolean `deleteDestinationIndex`, which if `true` then goes and deletes the index. As it is, someone could in theory pass a completely different index to the endpoint, which seems a bit dangerous. It might make sense to pass two parameters - `deleteDestinationIndex` and `deleteDestinationIndexPattern` as the modal does allow zero, one or both to be deleted. It would mean one extra query on the server side, to get the destination index for the job, but this approach would make more sense to me.

Reply:
You're right! This could be a security issue. I'll refactor it to use the boolean instead.

Reply:
Done in https://github.com/elastic/kibana/pull/66934/commits/d14fafab86d64ef3afce4bda0f3c415ea85e672e